### PR TITLE
API acceptance test for creating a share with user for matching files

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
As stated in issue #35414, two users having matching files/folders are made involved in a share and one of the user shares such matching files to another user or a group. API test scenario for such case is added.

## Related Issue

https://github.com/owncloud/core/issues/35414
All remaining checkboxes except sharing matching 
`files` to uninitialised user

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by kiranparajuli589** [comment]
@HariBhandari07 @skshetry @dpakach @jasson99

File: tests/acceptance/features/apiShareManagement/acceptShares.feature [hunk 1]
```diff
@@ -528,23 +528,34 @@ Feature: accept/decline shares coming from internal users
...
+    When user "user0" accepts the share "/PARENT" offered by user "user1" using the sharing API
+    And user "user0" declines the share "/PARENT (2)" offered by user "user1" using the sharing API
+    And user "user0" accepts the share "/PARENT" offered by user "user2" using the sharing API
+    And user "user0" accepts the share "/PARENT" offered by user "user1" using the sharing API
+    And user "user0" declines the share "/PARENT (2)" offered by user "user2" using the sharing API
+    And user "user0" declines the share "/PARENT (2) (2)" offered by user "user1" using the sharing API
+    And user "user3" shares folder "/PARENT" with user "user0" using the sharing API
+    And user "user0" accepts the share "/PARENT" offered by user "user3" using the sharing API
+    And user "user0" accepts the share "/PARENT" offered by user "user2" using the sharing API
+    And user "user0" accepts the share "/PARENT" offered by user "user1" using the sharing API
```
Comment (HariBhandari07) [inline tests/acceptance/features/apiShareManagement/acceptShares.feature]: Why are you removing and adding the same test again?

**Review by haribhandari07 — DISMISSED** [review]

File: tests/acceptance/features/apiShareManagement/acceptShares.feature [hunk 2]
```diff
@@ -528,23 +528,146 @@ Feature: accept/decline shares coming from internal users
...
+  Scenario: user shares matching folder with a user before that user has logged in
+    Given these users have been created with skeleton files but not initialized:
+      | username     |
+      | user3        |
+    And user "user0" uploads file with content "uploaded content" to "/PARENT/abc.txt" using the WebDAV API
+    When user "user0" shares folder "/PARENT" with user "user3" using the sharing API
+    Then user "user3" should see the following elements
+      | /PARENT/          |
+      | /PARENT/abc.txt   |
+    And user "user3" should not see the following elements
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareManagement/acceptShares.feature]: what happens in the step above?
when the test code does the API call for `Then user "user3" should see the following elements` that will cause `user3` to get initialized.
does the user get a `PARENT` skeleton folder called `PARENT (2)`? Or called something else? Or?

**Review by phil-davis — APPROVED** [review]
LGTM

File: tests/acceptance/features/apiShareManagement/acceptShares.feature (hunk 2 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/apiShareManagement/acceptShares.feature]: user doesn't get PARENT skeleton folder ` PARENT (2) `.
The next step below verifies that
`And the "user3" should not see the following elements`

Comment (kiranparajuli589) [reply tests/acceptance/features/apiShareManagement/acceptShares.feature]: I think the API call initializes the user and copies all the skeleton resources except the shared ones!

**Pull request merged**